penguin/tools/schema_contract.py

```
@@ -0,0 +1,155 @@
+"""Model-visible tool schema and runtime metadata contracts."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional
```

Ruff (0.15.12)

[warning] `typing.Dict` is deprecated, use `dict` instead

(UP035)